### PR TITLE
Give rollling upgrade tests more information

### PR DESCRIPTION
**Pull request by nik9000**
This passes the version that rolling upgrade tests are coming *from*
into the actual tests so they can reason about it. This is useful
because we have features that are not supported on early versions and
we want to write rolling upgrade tests for them. We can't run those
features or assert anything about them in when they don't exist. You'd
think we could use the minimum version of a node in the cluster, but
that only works in the unupgraded phases - once we've completed the
upgrade we need to have the version that we came from to know what we
did in the mixed version.

### CODE REVIEW

File: qa/rolling-upgrade/build.gradle [hunk 1]
```diff
@@ -42,6 +42,8 @@ for (Version bwcVersion : BuildParams.bwcVersions.wireCompatible) {
     }
   }
 
+  String oldVersion = bwcVersion.toString().replace('-SNAPSHOT', '')
```
Comment (mark-vieira) [inline qa/rolling-upgrade/build.gradle:45]: I don't think bwc versions will ever contain `-SNAPSHOT`, do they?

**Review by mark-vieira — APPROVED** [review]
One comment but otherwise LGTM.

File: qa/rolling-upgrade/build.gradle (hunk 1 above)
Comment (nik9000) [reply qa/rolling-upgrade/build.gradle:45]: I stole this from some other code. I'll check. If it doesn't I'll remove this and then zap it from the other spot too.

Comment (nik9000) [reply qa/rolling-upgrade/build.gradle]: Yeah - it doesn't have -SNAPSHOT. I've pushed an update that drops it from this PR and from the place where I found it.

**Comment by nik9000** [comment]
@elasticmachine, retest this please

**Comment by nik9000** [comment]
Backported with #76423

**Pull request merged**